### PR TITLE
Azure ml service

### PR DESCRIPTION
**Pull request by Blackmist**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Initial authoring

### CODE REVIEW

File: 101-azure-machine-learning-service-workspace/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Azure Machine Learning service workspace
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/README.md]: Use Title casing

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,134 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+            "type": "string",
+            "defaultValue": "myworkspace",
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: does this need to be globally unique?  If so, generate a unique name using uniqueString()

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            "metadata": {
+                "description": "The name of the Azure Machine Learning service workspace."
+            }
+        },
+        "workspaceLocation": {
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: rename this to "location"

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            "type": "string",
+            "defaultValue": "East US",
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: remove defaultValue

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            "apiVersion": "2018-02-14",
+            "location": "[parameters('workspaceLocation')]",
+            "properties": {
+                "tenantId": "[variables('keyVault').tenantId]",
+                "sku": {
+                    "name": "standard",
+                    "family": "A"
+                },
+                "accessPolicies": [],
+                "enabledForTemplateDeployment": true,
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: Are these flags are needed for this sample?  If not, remove them

File: 101-azure-machine-learning-service-workspace/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+            "type": "Microsoft.MachineLearningServices/workspaces",
+            "apiVersion": "2018-11-19",
+            "location": "[parameters('workspaceLocation')]",
+            "dependsOn": [
+                "[variables('storageAccount').name]",
+                "[variables('keyVault').name]",
+                "[variables('applicationInsightsName')]",
+                "[variables('containerRegistryName')]"
+            ],
+            "identity": {
```
Comment (bmoore-msft) [inline 101-azure-machine-learning-service-workspace/azuredeploy.json]: how is this used?

File: 101-azure-machine-learning-service-workspace/README.md (hunk 1 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/README.md]: Sadly, that is title casing for this service. The official service name is "Azure Machine Learning service", with a lower case 's' in 'service'. https://azure.microsoft.com/en-us/services/machine-learning-service/ is a marketing page where you can see the same capitalization.

The workspace is the central artifact for working with the service.

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 3 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: Will fix in a PR.

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 4 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: Will fix in next commit.

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 5 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: After checking, the accessPolicies is required, but the "enabled...." aren't. Will remove in next commit.

File: 101-azure-machine-learning-service-workspace/azuredeploy.json (hunk 6 above)
Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: Assuming you are asking about "identity". I've pinged engineering to get info on what it does. It's not documented in the reference docs, and the template errors if it's removed. Waiting on response from engineering.

Comment (Blackmist) [reply 101-azure-machine-learning-service-workspace/azuredeploy.json]: The identity element uses a managed system identity (MSI). It's required when creating the workspace through the template. More info is available at https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview. From reading that, it sounds like an AD entry is created to represent the service when it is created, then removed when you remove the service.

**Comment by Blackmist** [comment]
@bmoore-msft Made changes. Not sure why the CI build is failing though. The error doesn't indicate anything obvious. Using the same template via portal or Azure CLI works. I've even tried creating a parameters file with the randomly generated name from the logs, and it is working for me. Any advice on how to fix it?

**Comment by bmoore-msft** [comment]
There was a breaking change in a new version of node/npm - update your .travis.yml file to pick up that change or open a new PR after pulling from master.

**Pull request closed without merging**